### PR TITLE
RUC LSM update

### PR DESCRIPTION
**Pull request by tanyasmirnova**
This PR includes:
1. Adding leaf area index, wilting point and soil field capacity to the history files.
2. Adding a placeholder for the fire heat flux provided by smoke-dust code.
3. A change to the transpiration function to take into account the non-linear distribution of roots.

### CODE REVIEW

File: physics/lsm_ruc.F90 [hunk 1]
```diff
@@ -645,15 +649,38 @@ subroutine lsm_ruc_run                                            & ! inputs
       nsoil = lsoil_ruc
 
       do i  = 1, im ! i - horizontal loop
-        ! reassign smcref2 and smcwlt2 to RUC values
-        if(.not. land(i)) then
-          !water and sea ice
-          smcref2 (i) = one
-          smcwlt2 (i) = zero
+        ! reassign smcref2 and smcwlt2 to RUC values at kdt=1
```
Comment (grantfirl) [inline physics/lsm_ruc.F90]: Would this be clearer to put the `if (kdt==1)` in the else statement for `if (.not. land(i))`? It looks like the same thing is done for the non-land points whether or not it is the first time step.

File: physics/module_sf_ruclsm.F90 [hunk 2]
```diff
@@ -2049,6 +2068,9 @@ SUBROUTINE SFCTMP (debug_print, delt,ktau,conflx,i,j,         & !--- input varia
           eeta = eetas*(one-snowfrac) + eeta*snowfrac
           qfx = qfxs*(one-snowfrac) + qfx*snowfrac
           hfx = hfxs*(one-snowfrac) + hfx*snowfrac
+          !IF ( add_fire_heat_flux ) then ! JLS
```
Comment (grantfirl) [inline physics/module_sf_ruclsm.F90]: Any reason to keep this commented-out section here and in line 2119?

**Review by grantfirl — APPROVED** [review]
Looks OK to me with a couple minor comments.

File: physics/lsm_ruc.F90 (hunk 1 above)
Comment (tanyasmirnova) [reply physics/lsm_ruc.F90]: @grantfirl Agree, the change is made.

File: physics/module_sf_ruclsm.F90 (hunk 2 above)
Comment (tanyasmirnova) [reply physics/module_sf_ruclsm.F90]: @grantfirl The commented out section is removed.

**Review by Qingfu-Liu — APPROVED** [review]

**Review by Qingfu-Liu — COMMENTED** [review]
There are several lines of the code (physics/module_sf_ruclsm.F90) are commented out (lines 4083, 4086, 4540, 4543, 5217, 5220, 5793 and 5796, !fact = 5._kind_phys and !fact = 2._kind_phys), those lines can be removed.

**Comment by tanyasmirnova** [comment]
@Qingfu-Liu The commented lines are removed.

**Pull request merged**